sklearn/model_selection/tests/test_validation.py

```diff
@@ -1720,6 +1719,37 @@ def test_fit_and_score_working():
     assert result['parameters'] == fit_and_score_kwargs['parameters']
 
 
+def test_score_failing():
+    iris = load_iris()
+    X, y = iris.data, iris.target
+    clf = MockClassifier()
+
+    # the warning message we're expecting to see
+    warning_message = ("Scoring failed. The score on this train-test "
+                       "partition for these parameters will be set to %f. "
+                       "Details: \n" % np.nan)
```

Comment:
we should run this with two different values of error_score to make sure that it does what it's meant to